libs/sdk-core/src/moonpay/moonpay_url_signer.rs

```diff
@@ -0,0 +1,152 @@
+use anyhow::Result;
+use reqwest::Url;
+
+use crate::grpc::signer_client::SignerClient;
+use crate::grpc::SignUrlRequest;
+use crate::moonpay::moonpay_config::moonpay_config;
+
+#[tonic::async_trait]
+pub trait MoonPayUrlSigner: Send + Sync {
+    async fn sign_moon_pay_url(
+        &self,
+        moonpay_api_key: &str,
+        wallet_address: &str,
+        max_quote_currency_amount: &str,
+    ) -> Result<String>;
+}
+
+#[tonic::async_trait]
+impl MoonPayUrlSigner for SignerClient<tonic::transport::Channel> {
```

Comment:
Seems not related. Why do we need this implementation for the signer?  We are using breez server to sign don't we?

Reply:
Yes, we are using the breez server. Under the hood, its SignerClient is the one signing, so I thought having this trait would be a good thing.

Reply:
I started removing the `MoonPayUrlSigner`, but I got a problem; without this layer, I cannot unit test unless changing how the breez server is made by adding a trait to `get_signer_client` and implementing my version of it on the mocks version. What do you prefer? The `MoonPayUrlSigner` for the `SignerClient` or a `SignerClientProvider` for the Breez server? Or another approach I'm not aware of.

Reply:
Not because the `get_signer_client(&self)`, I'll implement it so you can check what I'm talking about

Reply:
No success; I couldn't find a way to unit test without adding this layer.

Reply:
The test I'm talking about is this one: https://github.com/breez/breez-sdk/pull/189/files#diff-a20deab6b2aa49948975cb34e64974e4fef825884eb03b024717519c82d410ceR72-R131

I'm creating a mock web server to simulate the URL signature, so I need a layer or mechanism to provide a mock SignerClient.

I've succeeded in running the test without this layer, but the test was executing a real request to the breez server, which is a bad idea for a unit test. Ideally, we mock the server or mock the client.

Do you see another way of doing it?